Add environment variables for BDBag-to-S3 to dcc-dashboard-service (resolves #61)

mjkrause opened the pull request:
Added `AZUL_S3_AWS_REGION` and `AZUL_PRESIGNED_URL_EXPIRATION` to `dcc-dashboard-service`, the Boardwalk mustache template, and to `dev.yml` and `prod.yml` in the boardwalk directory.

I ran `install_bootstrap` on a VM. You can test it in a terminal with 
`curl -v "https://mkrause4.ucsc-cgp-dev.org/api/v1/repository/files/export?filters=%7B%7D&format=tsv" -H "Content-Type: application/json"`

Commit:
Added aws_s3_aws_region to install_bootstrap

Commit:
Added aws_s3_aws_region to install_bootstrap

Commit:
Added azul_presigned_url_expiration to install_bootstrap

Commit:
Added azul_presigned_url_expiration to install_bootstrap

Commit:
Added vars azul_s3_aws_region and azul_presigned_url_expiration to dev.yml

Commit:
Added vars azul_s3_aws_region and azul_presigned_url_expiration to dev.yml

Commit:
Added vars azul_s3_aws_region and azul_presigned_url_expiration to prod.yml

Commit:
Added vars azul_s3_aws_region and azul_presigned_url_expiration to prod.yml

Commit:
Added vars azul_s3_aws_region and azul_presigned_url_expiration to boardwalk mustache template

Commit:
Added vars azul_s3_aws_region and azul_presigned_url_expiration to boardwalk mustache template

Commit:
Removed superfluous spaces

Commit:
Removed superfluous spaces

Commit:
Removed superfluous spaces

Commit:
Removed superfluous spaces

Commit:
Forgot to set correct value names for parameters

Commit:
Forgot to set correct value names for parameters

Commit:
Added "AWS" to the BDBag-S3 region question

Commit:
Added "AWS" to the BDBag-S3 region question

coverbeck commented:
Since you are in here anyway, can you please remove the firecloud-bagit-lambda questions and variables?

coverbeck commented on install_bootstrap:
```diff
@@ -285,6 +285,9 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         ask_question "What is your AWS region?" "$AWS_DEFAULT_REGION" "AWS region" $aws_default_region
 
+        azul_s3_aws_region='azul_s3_aws_region'
+        ask_question "What is the AWS region where the BDBag for export to FireCloud will be temporarily be stored?" "$AZUL_S3_AWS_REGION" "Azul_S3_Aws_Region" $azul_s3_aws_region
```
`be` twice

mjkrause replied:
Fixed.

coverbeck commented on install_bootstrap:
```diff
@@ -323,6 +326,8 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         ask_question "What is the name of the AWS S3 bucket to store BDBags?" "$AZUL_S3_BUCKET" "Azul S3 Bucket" $azul_s3_bucket
 
+	azul_presigned_url_expiration='azul_presigned_url_expiration'
+	ask_question "Choose an expiration time (in seconds) for presigned URL to the S3-BDBag for FC export (recommended 3600)?" "$AZUL_PRESIGNED_URL_EXPIRATION" "Azul Presigned Url Expiration" $azul_presigned_url_expiration
```
If you keep this, either `What is...?`, or `Choose...` with no question mark at the end. But I don't think you need this question...  I think you should hard-code the value like `es_port` and `es_protocol` are hard-coded with the sensible default you have in your code.

My point is that a user will typically not need to change this, just like they don't need to change the `es_port`. But on the off chance that they do, we will have an environment variable than can be modified without changing the code.

mjkrause replied:
I followed your suggestion, and removed the question.

mjkrause commented:
> Since you are in here anyway, can you please remove the firecloud-bagit-lambda questions and variables?

How about we do that when working on https://ucsc-cgl.atlassian.net/browse/TLP-560? It seems a better packaging of the tasks.

Commit:
Corrected grammar in question

Commit:
Corrected grammar in question

Commit:
azul_presigned_url_expiration is set in the script, but user is not asked for it anymore

Commit:
azul_presigned_url_expiration is set in the script, but user is not asked for it anymore

jessebrennan commented on install_bootstrap:
```diff
@@ -285,6 +285,9 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         ask_question "What is your AWS region?" "$AWS_DEFAULT_REGION" "AWS region" $aws_default_region
 
+        azul_s3_aws_region='azul_s3_aws_region'
+        ask_question "What is the AWS region where the BDBag for export to FireCloud will temporarily be stored?" "$AZUL_S3_AWS_REGION" "Azul_S3_Aws_Region" $azul_s3_aws_region
```
This is also the same region as something else, right? What else?

jessebrennan replied:
Just me personally following these instructions, I would have no clue what to put for this. Is there more info that can be added to the question to make it more clear?

coverbeck commented on install_bootstrap:
```diff
@@ -323,6 +326,7 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         azul_s3_bucket='azul_s3_bucket'
         ask_question "What is the name of the AWS S3 bucket to store BDBags?" "$AZUL_S3_BUCKET" "Azul S3 Bucket" $azul_s3_bucket
 
+	azul_presigned_url_expiration='azul_presigned_url_expiration'
```
This should be `azul_presigned_url_expiration=3600`, or whatever the default value is supposed to be.

mjkrause replied:
fixed

jessebrennan commented on install_bootstrap:
```diff
@@ -323,6 +326,7 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         azul_s3_bucket='azul_s3_bucket'
         ask_question "What is the name of the AWS S3 bucket to store BDBags?" "$AZUL_S3_BUCKET" "Azul S3 Bucket" $azul_s3_bucket
 
+	azul_presigned_url_expiration='azul_presigned_url_expiration'
```
I'm sort of confused, it looks like you never ask for the resigned url expiration. It's just supposed to be set to this string?

mjkrause replied:
fixed

jessebrennan commented:
Also I tried your curl command and it works great except the output I got was not json even though that's what the content-type is set to

coverbeck commented on install_bootstrap:
```diff
@@ -285,6 +285,9 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         ask_question "What is your AWS region?" "$AWS_DEFAULT_REGION" "AWS region" $aws_default_region
 
+        azul_s3_aws_region='azul_s3_aws_region'
+        ask_question "What is the AWS region where the BDBag for export to FireCloud will temporarily be stored?" "$AZUL_S3_AWS_REGION" "Azul_S3_Aws_Region" $azul_s3_aws_region
```
I would leave `FireCloud` out of it, and it should be plural.

Thinking about it more, because this applies to the azul_s3_bucket, I think you should move this below line 327, where you ask for the name of azul bucket, and simply ask something like`What is the AWS region of the S3 <bucket name user entered in answering previous question> bucket?`. Or if that's too tricky, reference the previous question more generically.

mjkrause replied:
fixed

coverbeck committed:
Remove endpoint variable being passed to Boardwalk (fixes #64) (#65)

jessebrennan committed:
Add aws region to azul

This is neede for elastic search authorization

jessebrennan committed:
Merge pull request #66 from DataBiosphere/feature/add-region-azul

Add aws region to azul

jessebrennan committed:
Update images for production

This integrates the latest releases on the different components.

jessebrennan committed:
Merge pull request #67 from DataBiosphere/feature/update-prod-images

Update images for production

jessebrennan committed:
Update docker images for azul and dashboard

jessebrennan committed:
Merge pull request #70 from DataBiosphere/feature/update-azul

Update docker images for azul and dashboard

Commit:
Changed position of region question and added default expiration time

The question for the default AWS region of the azul-s3-bucket now follows now
right on the heels of the question for the name of the azul-s3-bucket. That way
the significance of that question should become more obvious.

I assigned 3600 s as the default value for the presigned URL of the BDBag
in the azul-s3-bucket.

Commit:
Changed position of region question and added default expiration time

The question for the default AWS region of the azul-s3-bucket now follows now
right on the heels of the question for the name of the azul-s3-bucket. That way
the significance of that question should become more obvious.

I assigned 3600 s as the default value for the presigned URL of the BDBag
in the azul-s3-bucket.

Commit:
Rephrased question re. AWS region for dcc-dashboard

Although this isn't part of the story, it will illuminate the difference
between the region of the VM which has deployed the backend and the region in
which the S3 bucket resides that stores temporary BDBags.

Commit:
Rephrased question re. AWS region for dcc-dashboard

Although this isn't part of the story, it will illuminate the difference
between the region of the VM which has deployed the backend and the region in
which the S3 bucket resides that stores temporary BDBags.

Commit:
Merge branch 'feature/mkrause/61-add-env-vars' of github.com:DataBiosphere/cgp-deployment into feature/mkrause/61-add-env-vars

This includes changes made on feature/commons in while I developed on this
feature branch.

Commit:
Correct grammar in AWS region for dcc-dashboard

mjkrause commented:
I rebased the branch on `feature/commons` to reflect latest changes Jesse put it in the meantime.

I addressed all comments and suggestions. I've created a new instance, which you can test at
`curl -v "https://mkrause5.ucsc-cgp-dev.org/api/v1/repository/files/export?filters=%7B%7D&format=tsv"
` for the TSV, or
`curl -v "https://mkrause5.ucsc-cgp-dev.org/api/v1/repository/files/export?filters=%7B%7D&format=tsv"
`
to return the signed URL for the BDBag as the value in JSON.

Inspired by Jesse's concern that the user may not know how to answer the question for the AWS region for the S3 bucket to store the BDBags I thought it was a good idea to [rephrase an earlier question for the AWS region for the EC2 dcc-dashboard backend](https://github.com/DataBiosphere/cgp-deployment/pull/62/commits/416d9b349c1ca313c60988aed27736ba1e43fd0b).

coverbeck commented on boardwalk/dev.yml:
```diff
@@ -100,7 +103,6 @@ services:
     build: boardwalk
     environment:
       - BOARDWALK_HOST=${dcc_dashboard_host}
-      - BW_DATA_URL=${dcc_dashboard_protocol}://${dcc_dashboard_host}
```
This should not be showing up as change. Can you please resolve it so it doesn't.

mjkrause replied:
We decided not to worry about this as the branch merges with `feature/commons`.

coverbeck commented on boardwalk/prod.yml:
```diff
@@ -21,7 +24,7 @@ services:
       FC_LAMBDA_PORT: "${fc_lambda_port}"
       DOS_DSS_SERVICE: "${dos_dss_service}"
-    image: quay.io/ucsc_cgl/azul-commons:commons-0.1.5-beta
+    image: quay.io/ucsc_cgl/azul-commons:commons-0.1.9-beta
```
This should not be showing up as change. Can you please resolve it so it doesn't.

mjkrause replied:
We decided not to worry about this as the branch merges with `feature/commons`.

coverbeck commented on boardwalk/prod.yml:
```diff
@@ -37,7 +40,7 @@ services:
     restart: always
   dcc-dashboard:
-    image: quay.io/ucsc_cgl/commons-dashboard:commons-0.1.3-beta
+    image: quay.io/ucsc_cgl/commons-dashboard:commons-0.1.5-beta
```
This should not be showing up as change. Can you please resolve it so it doesn't.

mjkrause replied:
We decided not to worry about this as the branch merges with `feature/commons`.

coverbeck commented on boardwalk/prod.yml:
```diff
@@ -96,10 +99,9 @@ services:
     restart: always
   boardwalk:
-    image: quay.io/ucsc_cgl/commons-boardwalk:commons-0.1.1-beta
+    image: quay.io/ucsc_cgl/commons-boardwalk:commons-0.1.2-beta
```
This should not be showing up as change. Can you please resolve it so it doesn't.

mjkrause replied:
We decided not to worry about this as the branch merges with `feature/commons`.

coverbeck commented on boardwalk/prod.yml:
```diff
@@ -96,10 +99,9 @@ services:
+    image: quay.io/ucsc_cgl/commons-boardwalk:commons-0.1.2-beta
     environment:
       - BOARDWALK_HOST=${dcc_dashboard_host}
-      - BW_DATA_URL=${dcc_dashboard_protocol}://${dcc_dashboard_host}
```
This should not be showing up as change. Can you please resolve it so it doesn't.

mjkrause replied:
We decided not to worry about this as the branch merges with `feature/commons`.

coverbeck commented on install_bootstrap:
```diff
@@ -283,7 +283,7 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         ############################
         aws_default_region='aws_default_region'
-        ask_question "What is your AWS region?" "$AWS_DEFAULT_REGION" "AWS region" $aws_default_region
+        ask_question "What AWS region does dcc-dashboard reside in?" "$AWS_DEFAULT_REGION" "AWS region" $aws_default_region
```
This region is for used for the location of the secrets manager. It's good you  reworded this, but I think you should say `What AWS region does the AWS Secrets Manager reside in?`.

mjkrause replied:
done

coverbeck commented on install_bootstrap:
```diff
@@ -314,15 +314,25 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         ask_question "What is the name of the AWS S3 bucket to store BDBags?" "$AZUL_S3_BUCKET" "Azul S3 Bucket" $azul_s3_bucket
 
+        azul_s3_aws_region='azul_s3_aws_region'
+        ask_question "What is the AWS region of that AWS S3 bucket?" "$AZUL_S3_AWS_REGION" "Azul_S3_Aws_Region" $azul_s3_aws_region
```
👍

coverbeck commented on install_bootstrap:
```diff
@@ -314,15 +314,25 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
         google_site_verification_code='google_site_verification_code'
         ask_question "What is your Google site verification code, something like tk74JOp1VBqdRQVTirM8WKpa1VIYW_vwxfoJo53inH8?" "$GOOGLE_SITE_VERIFICATION_CODE" "Google Site Verification Code" $google_site_verification_code
 
+	# bagit-firecloud-lambda 
```
I know we discussed this and you disagree, but I think you should get rid of the bagit stuff in this PR. This PR finalizes the completion of the presigned url, and it's weird to be prompted for the Bagit info when it's not being used.

mjkrause replied:
Removed all variables referring to the `bagit-firecloud-lambda`, brought the docker compose system down, then up again, and tested the endpoint. I did not go through the installation process.

Commit:
Rephrasing question in installer and removing bagit lambda vars

Rephrased corrected one of the questions for the AWS region to refer
to the AWS Secrets Manager.

In addition I removed all references to variables for the
bagit-firecloud-lambda.

coverbeck reviewed (changes requested):
Can you please remove appropriate lines starting at 393 as well.

After you have done that, can you squash all your commits into one and push that? Given the slightly odd behavior we are seeing with the diff, I would like to see how it looks with one commit.

Thanks.

Commit:
Removed bagit lambda vars from configuration dict
